src/pysonar_scanner/jre.py

```diff
@@ -102,7 +102,12 @@ def __download_jre(self, jre: JRE) -> Optional[pathlib.Path]:
         cache_file.filepath.unlink(missing_ok=True)
 
         with cache_file.open(mode="wb") as f:
-            self.api.download_analysis_jre(jre.id, f)
+            if jre.download_url is not None:
+                self.api.download_file_from_url(jre.download_url, f)
+            elif jre.id is not None:
+                self.api.download_analysis_jre(jre.id, f)
+            else:
+                raise JreProvisioningException("JRE ID and download URL are both None")
```

Comment:
I think this is not an explicit error message for a lambda user. I propose:
```suggestion
                raise JreProvisioningException("Failed to download the JRE using SonarQube. If this problem persists, you can use the option --sonar-scanner-java-exe-path to use your own local JRE.")
```